apps/web/src/lib/job/job-search.ts

```diff
@@ -15,6 +15,12 @@ export function jobMatchesQuery(job: SearchableJob, query: string): boolean {
   if (!query) return true;
   const term = query.toLowerCase();
 
+  // Extract links from events if available (for JobWithStatus), otherwise use direct links property
+  const links =
+    "events" in job && Array.isArray(job.events)
+      ? job.events.flatMap((event) => event.links ?? [])
+      : job.links ?? [];
+
```

Comment:
Here I would also recommend to provide this via a function in the service layer or via a computed field in the from the repo-layer.